File: heclib/heclib_f/src/DssInterface/checks.f
     
      subroutine checkString(stringOrig, stringRead, mess, status)
     
      implicit none
     
      integer status
      character*(*) stringOrig, stringRead
      character mess*(*)
      character strOrig*200, stringR*200

        if (stringOrig.ne.stringRead) then
!       Allow different case strings
        strOrig = stringOrig
        stringR = stringRead
        call UPCASE(strOrig)
        call UPCASE(stringR)
        if (strOrig.ne.stringR) then
         write (*,*)' '
         write (*,*)' '
         write (*,*)'***  String read does not match that written *****'
         write (*, 20) stringOrig, stringRead
20       Format('String Written: ==>', A, '<==  Read: ==>',A, '<==')
         write (*,*)mess
         write (*,*)' '
         write (*,*)' '
         status = -1
         return
        endif
        endif
 100  continue

      status = 0
      return
      end
     
     
     
      subroutine checkTimes(dataOrig, dataRead, baseDate,
     * timeGranularityFlag, number, mess, status)
   
      implicit none
   
      integer number, status
      integer dataOrig(number), dataRead(number)
      character mess*(*)
   
      integer i, minutesInBaseDate, baseDate
      integer timeRead, timeGranularityFlag
   
      minutesInBaseDate = baseDate * 1440
      do 100 i=1, number
        if (timeGranularityFlag.eq.1) then
          timeRead = (dataRead(i) / 60) + minutesInBaseDate
        else
          timeRead = dataRead(i) + minutesInBaseDate
        endif
        if (dataOrig(i).ne.timeRead) then
        write (*,*)' '
        write (*,*)' '
        write (*,*)'*****  Data read does not match those written *****'
        write (*, 20) i, dataOrig(i), timeRead
 20     Format('At ordinate: ', I8, ' Written: ', I12, '  Read: ',I12)
         write (*,*)mess
         write (*,*)' '
         write (*,*)' '
        status = -1
        return
        endif
 100  continue

      status = 0
      return
      end


      subroutine checkNotes(dataOrig, dataRead, number, mess, status)
   
      implicit none
   
      integer number, status
      character dataOrig(number)*(*), dataRead(number)*(*)
      character mess*(*)
   
      integer i, n2, n3

      do 100 i=1, number
        call chrlnb(dataOrig(i), n3)
        call chrlnb(dataRead(i), n2)
        if ((n3.le.0).and.(n2.le.0)) go to 100
        if (n3.le.0) n3 = 1
        if (n2.le.0) n2 = 1
        call checkString(dataOrig(i)(1:n3), dataRead(i)(1:n2), mess,
     * status)
        if (status.ne.0) then
            write(*,*)'At ordinate number ',i
            status = -1
            return
        endif
 100  continue

      status = 0
      return
      end
   
      subroutine checkBlanks(dataRead, number, mess, status)
   
      implicit none
   
      integer number, status
      character dataRead(number)*(*)
      character mess*(*)
   
      integer i, n2

      do 100 i=1, number
      call chrlnb(dataRead(i), n2)
      if (n2.ne.0) then
      write (*,*)' '
      write (*,*)' '
      write (*,*)'***** Strings read does not match those written *****'
       write (*,*)mess
       write (*,*)' '
       write (*,*)' '
      write(*,*)'At ordinate number ',i
      status = -1
      return
      endif
 100  continue

      status = 0
      return
      end

File: heclib/heclib_c/src/Utilities/checks.c
#include "checks.h"
#include "heclib7.h"

void checkdoubles_(double* dataOrig, double* dataRead, int* number, const char* mess, int* status, size_t dummy) {
	int i;

	*status = 0;

	for (i = 0; i < *number; i++) {
		if (dataOrig[i] != dataRead[i]) {
			printf(" \n");
			printf(" \n");
			printf(" *****  Data read does not match those written  *****\n");
			printf(" At ordinate: %8d Written: %12.3f  Read: %12.3f\n", i + 1, dataOrig[i], dataRead[i]);
			printf(" %s\n", mess);
			printf(" \n");
			printf(" \n");
			*status = -1;
			return;
		}
	}

	return;
}


void checknumbers_(int* numberOrig, int* numberRead, const char* mess, int* status, size_t dummy) {
	if (*numberOrig != *numberRead) {
		printf(" \n");
		printf(" \n");
		printf(" *****  Number read does not match that written  *\n");
		printf(" Number Written: %10d  Read: %10d\n", *numberOrig, *numberRead);
		printf(" %s\n", mess);
		printf(" \n");
		printf(" \n");
		*status = -1;
	}
	else {
		*status = 0;
	}
}

void checkfloats_(float* dataOrig, float* dataRead, int* number, const char* mess, int* status, size_t dummy) {
	int i;

	*status = 0;

	for (i = 0; i < *number; i++) {
		if (dataOrig[i] != dataRead[i]) {
			printf(" \n");
			printf(" \n");
			printf(" *****  Data read does not match those written  *****\n");
			printf(" At ordinate: %8d Written: %12.3f Read: %12.3f\n", i + 1, dataOrig[i], dataRead[i]);
			printf(" %s\n", mess);
			printf(" \n");
			printf(" \n");
			*status = -1;
			return;
		}
	}
}

void checkints_(int* dataOrig, int* dataRead, int* length, int* number, const char* mess, int* status, size_t mess_len)
{
	int i, j;
	for (i = 0; i < *number; i++) {
		for (j = 0; j < *length; j++) {
			if (dataOrig[i * (*length) + j] != dataRead[i * (*length) + j]) {
				printf(" \n\n");
				printf(" *****  Data read does not match those written *****\n");
				printf(" At ordinate: %8d Written: %12d Read: %12d\n",
					i + 1, dataOrig[i * (*length) + j], dataRead[i * (*length) + j]);
				printf(" %s\n", mess);
				printf(" \n\n");
				*status = -1;
				return;
			}
		}
	}
	*status = 0;
}


void checkstring_(char* stringOrig, char* stringRead, char* mess, int* status,
	size_t stringOrigLen, size_t stringReadLen, size_t messLen) {
	int max_len = 201;
	char strOrig[201], stringR[201];
	

	if (strncmp(stringOrig, stringRead, stringOrigLen) != 0) {
		// Allow different case strings
		stringCopy(strOrig, max_len, stringOrig, stringOrigLen);
		stringCopy(stringR, max_len, stringRead, stringReadLen);

		upperCase(strOrig);
		upperCase(stringR);

		if (strcmp(strOrig, stringR) != 0) {
			printf(" \n\n");
			printf(" ***  String read does not match that written *****\n");
			printf(" String Written: ==>%s<==  Read: ==>%s<==\n", stringOrig, stringRead);
			printf(" %s\n", mess);
			printf(" \n\n");
			*status = -1;
			return;
		}
	}

	*status = 0;
}
